nomad/structs/structs.go

```diff
@@ -5666,11 +5685,27 @@ func CronParseNext(e *cronexpr.Expression, fromTime time.Time, spec string) (t t
 func (p *PeriodicConfig) Next(fromTime time.Time) (time.Time, error) {
 	switch p.SpecType {
 	case PeriodicSpecCron:
-		e, err := cronexpr.Parse(p.Spec)
-		if err != nil {
-			return time.Time{}, fmt.Errorf("failed parsing cron expression: %q: %v", p.Spec, err)
+		if p.Spec != "" {
+			return CronParseNext(fromTime, p.Spec)
+		}
+
+		times := make([]time.Time, len(p.Specs))
+		var nextTime time.Time
+		var err error
+		for i, spec := range p.Specs {
+			times[i], err = CronParseNext(fromTime, spec)
+			if err != nil {
+				return times[i], err
+			}
 		}
-		return CronParseNext(e, fromTime, p.Spec)
+		nextTime = times[0]
+		for _, next := range times {
+			if next.Before(nextTime) {
+				nextTime = next
+			}
+		}
```

Comment:
The same update from `api/jobs.go` can be applied here to avoid the second iteration and storage.

Reply:
This was missed as well 😄